### PR TITLE
Alternative tsconfig.json

### PR DESCRIPTION
**Pull request by Pirosauro**
See https://github.com/privatenumber/pkgroll/issues/36

- added the `--tsconfig` CLI flag
- the tsconfig is now passed as parameter to esbuild

### CODE REVIEW

File: src/utils/get-rollup-configs.ts [hunk 1]
```diff
@@ -58,9 +59,11 @@ const getConfig = {
 		aliases: AliasMap,
 		env: EnvObject,
 		executablePaths: string[],
+		tsconfigRaw: TsConfigJsonResolved
 	) {
 		const esbuildConfig = {
 			target: options.target,
+			tsconfigRaw: tsconfigRaw as string,
```
Comment (privatenumber) [inline src/utils/get-rollup-configs.ts:66]:
```suggestion
			tsconfigRaw: tsconfigRaw as TransformOptions['tsconfigRaw'],
```

File: src/cli.ts [hunk 2]
```diff
@@ -96,6 +102,7 @@ const cwd = process.cwd();
  */
 const sourcePath = normalizePath(argv.flags.src, true);
 const distPath = normalizePath(argv.flags.dist, true);
+const tsconfig = getTsconfig(undefined, argv.flags.tsconfig);
```
Comment (privatenumber) [inline src/cli.ts:105]: `parseTsconfig` should be used here. `getTsconfig` searches for a tsconfig.json file, but `parseTsconfig` parses the tsconfig.json at a given path.

Example:
https://github.com/esbuild-kit/cjs-loader/blob/94a69f9e6d41225cb8a132938a3b9b077c15f966/src/index.ts#L23

**Comment by privatenumber** [comment]
Thanks for the PR! Can you add a test?

File: src/cli.ts [hunk 3]
```diff
@@ -96,8 +102,9 @@ const cwd = process.cwd();
  */
 const sourcePath = normalizePath(argv.flags.src, true);
 const distPath = normalizePath(argv.flags.dist, true);
+const tsconfig = parseTsconfig(argv.flags.tsconfig);
```
Comment (privatenumber) [inline src/cli.ts:105]: While `parseTsconfig` should be used if a tsconfig file is passed in, it should default to `getTsconfig` if it's not passed in.

For reference, it should look like this:
https://github.com/esbuild-kit/esm-loader/blob/332a0a0220dbb80de806ebb3a15ec7c3ce893bee/src/utils.ts#L13-L20

**Comment by error-four-o-four** [comment]
hey there,

I wonder if @Pirosauro is still working on this PR? Otherwise I'd like to be assigned

Cheers

**Comment by privatenumber** [comment]
Fine by me! Feel free to branch off of `Pirosauro:pkgroll-36` and open a new PR

**Pull request closed without merging**